core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -900,6 +899,57 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    def handleUnauthorizedRequest = {
```

Comment:
We can just throw AuthorizationException, which will then be handled in CreateTopicRequest.getErrorResponse().